src/Rings/msolve/f4.jl

```diff
@@ -101,8 +96,7 @@ function f4(
         basis = convert_ff_gb_array_to_singular_ideal(
           jl_ld, jl_len, jl_exp, jl_cf, R)
     # end
-    sym = Libdl.dlsym(lib, :free_julia_data)
-    ccall(sym, Nothing , (Ptr{Ptr{Cint}}, Ptr{Ptr{Cint}}, Ptr{Ptr{Cvoid}},
+    ccall((:free_julia_data, libneogb), Nothing , (Ptr{Ptr{Cint}}, Ptr{Ptr{Cint}}, Ptr{Ptr{Cvoid}},
                 Int, Int), gb_len, gb_exp, gb_cf, jl_ld, field_char)
     # free data
     ccall(:free, Nothing , (Ptr{Cint}, ), gb_ld)
```

Comment:
So, if you use the julia ref for gb_ld the malloc above and this free here can go away.

Reply:
Also, the code `= Nothing` (Nothing is a type) below for "for letting the garbage collector free memory" should never be needed as the references to the object are only local to this function.

Reply:
OK, will try to do so. Can I do something related with the `Ptr{Ptr{Cint}}` stuff?

Reply:
If the code between the malloc/free is bulletproof, i.e. can't be safely aborted and can't throw, then the malloc/free is alright, thought not very readable julia. If the code between can do arbitrary things, then wrapping all the pointers together in one `mutable struct` and attaching an appropriate finalizer will allow the GC to clean it up when the code between throws.
I will look at this specific case soon.

Reply:
This is discussion is subsumed in the one on line 65.